src/ui/blocks/RemoteSource.hpp

```diff
@@ -11,6 +11,8 @@
 #include <RestClient.hpp>
 #include <type_traits>
 
+#include "../app.h"
```

Comment:
Looks wrong to me that the source now depends on the main header, I'd suspect that in the end this will even be a cyclic dependency (caught by the include guard of course). This makes it a lot more difficult to follow the code and also makes it impossible to unit-test this part of the code independently from the rest of the repository as this header will basically pull in the rest of the repository.

The app should just get all remote blocks from the graph to get the URLs.

Reply:
I agree, I think the goal should be to use RemoteSource as a normal block in a non-UI GR context, and maybe even move it out of this repo (gr-digitizers).
Once Ivan's settings msg patches are in, this could rely on initial iterating the blocks in the graph, and then listening to settings update messages. My DNS patch (https://github.com/fair-acc/opendigitizer/pull/168)  will do something similar in the service, to learn about signal metadata changes.

Reply:
As discussed, decided that this will require additional restructuring of the project in general which is out of scope for this PR but should be tackled rather soon to get openditigizer into a state that is more inviting to understand/contribute/maintain. // cc @giucam  @ivan-cukic